driver/razermouse_driver.c

```diff
@@ -1419,7 +1537,30 @@ static ssize_t razer_attr_read_logo_led_effect(struct device *dev, struct device
 }
 
 /**
- * Write device file "mode_spectrum" (for extended mouse matrix effects)
+ * Write device file "scroll_mode_wave" (for extended mouse matrix effects)
+ *
+ * Wave effect mode is activated whenever the file is written to
+ */
+static ssize_t razer_attr_write_scroll_mode_wave(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
+{
+    struct usb_interface *intf = to_usb_interface(dev->parent);
+    struct usb_device *usb_dev = interface_to_usbdev(intf);
+    unsigned char direction = (unsigned char)simple_strtoul(buf, NULL, 10);
+    struct razer_report report = {0};
+
```

Comment:
clamp stuff here?

Reply:
Clamped to match other values between 0x00 and 0x02 inclusive as Razer can't decide what they want.  At a minimum we would need 1 and 2 here for this mouse.  If you prefer that let me know.

Reply:
Very open to constructive criticism or advice if you want me to modify anything.